### DISCUSSION TITLE
Unresponsiveness initial_hvac_mode: "off"

### DISCUSSION THREAD

**Discussion by Stefano0042** [ScratMan/HASmartThermostat#37] [Ideas]
Despite initial hvac_mode: is set to "off" after every new ha restart it remain on the old state.
The same issue is involving target_temp: value.

Restart Home Assistant
OS: Raspberry PI OS
All the browser
HA ver.: core-2021.11.5

**Comment by ScratMan** [comment]
Hello, this is on purpose, the "restore" feature of the integration is called after HA restart to set all the main parameters to their last known states. Otherwise in case of HA crash with a restart while you're away the heating system would be stuck to off. This setting is used only when there is no old state data in the database.

If you really need the thermostat to be forced to off (no regulation, no heating) after HA restart, you may use an automation triggered by the HA boot event.

**Reply by Stefano0042** [reply]
Hello Adrien, It could be an acceptable idea to create a dev branch without "restore" for 
initial_hvac_mode:
target_temp:
....
Under the umbrella of "keep it simple" I appreciate the Generic Thermostat behaviour, thus the possibility to restart in "off" mode. My "target sensor:" is a collection of Zigbee temperature sensors, and take some time to build up the true current value. By starting in "off" mode unnecessarily heat calls are avoided.

**Reply by Stefano0042** [reply]
Or even better, keeping both the possibilities alive by an extra parameter e.g. old_state_restore:  (boolean)

**Reply by ScratMan** [reply]
Hello @Stefano0042, please try this beta release : https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.12.2-beta1

**Reply by Stefano0042** [reply]
Hello @ScratMan, I like this way, brilliant solution! I wonder if we can manage "target_temp: " the same way ( ...rhetorical question :):)  ).

**Reply by Stefano0042** [reply]
Hi @ScratMan, I tried to adjust "target_temp:" by myself, but as it is a mix of Generic Thermostat bunch of inherited program lines and new programming, is not so easy to deal with at my current status of python knowledge.

**Reply by ScratMan** [reply]
Try [v2021.12.2-beta3](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.12.2-beta3) 😉

**Reply by Stefano0042** [reply]
Hi @ScratMan I've got it (-352... 354). It works as expected! Thank you again.

**Comment by Stefano0042** [comment]
Hello, thank you for your fast reply, but it seem to be a misconception on my eyes, if we want the heating to be on after a restart or crash:  
initial_hvac_mode: "heat"
target_temp: ...desired temp °C
